idaes/models_extra/power_generation/unit_models/soc_submodels.py

```diff
@@ -0,0 +1,3997 @@
+        )
+        CONFIG.declare(
+            "Dconc_x1",
+            ConfigValue(default=None,
+                        description="Deviation of concentration at x= 0 from "
+                        "channel bulk concentration"),
+        )
+        CONFIG.declare(
+            "xflux_x0",
+            ConfigValue(default=None, description="Variable for material flux at x=0"),
+        )
+        CONFIG.declare(
+            "xflux_x1",
+            ConfigValue(default=None, description="Variable for material flux at x=1"),
+        )
+        
+def _create_material_boundary_conditions_if_none(unit,thin):
+    tset = unit.flowsheet().config.time
+    iznodes = unit.iznodes
+    comps = unit.comps
+    if thin:
+        _create_if_none(unit, 
+                        "Dconc",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/pyo.units.m**3)
+        _create_if_none(unit, 
+                        "xflux",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/(pyo.units.s*pyo.units.m**2))
+    else:
+        _create_if_none(unit, 
+                        "Dconc_x0",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/pyo.units.m**3)
+        _create_if_none(unit, 
+                        "Dconc_x1",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/pyo.units.m**3)
+        _create_if_none(unit, 
+                        "xflux_x0",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/(pyo.units.s*pyo.units.m**2))
+        _create_if_none(unit, 
+                        "xflux_x1",
+                        idx_set=(tset,iznodes,comps),
+                        units=pyo.units.mol/(pyo.units.s*pyo.units.m**2))
+    
+@declare_process_block_class("SocChannel")
+class SocChannelData(UnitModelBlockData):
+    CONFIG = ConfigBlock()
+    CONFIG.declare(
+        "dynamic",
+        ConfigValue(
+            domain=In([useDefault, True, False]),
+            default=useDefault,
+            description="Dynamic model flag",
+            doc="""Indicates whether this model will be dynamic,
+                **default** = useDefault.
+                **Valid values:** {
+                **useDefault** - get flag from parent (default = False),
+                **True** - set as a dynamic model,
+                **False** - set as a steady-state model.}""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(domain=In([useDefault, True, False]), default=useDefault),
+    )
+    CONFIG.declare(
+        "comp_list",
+        ConfigValue(default=["H2", "H2O"], description="List of components"),
+    )
+    
+    CONFIG.declare(
+        "cv_zfaces",
+        ConfigValue(description="CV boundary set, should start with 0 and end with 1."),
```

Comment:
This looks like it should have a domain of some sort to make sure you get the type of input you are expecting. I think you should also be clearer about exactly what form the input should be in the doc string (is this a Python `list`, or a Pyomo `Set`)?

This applies to other config arguments below.